go/vt/vtgate/engine/merge.go

```diff
@@ -0,0 +1,72 @@
+package engine
+
+import (
+	"sync"
+
+	"github.com/youtube/vitess/go/sqltypes"
+	"github.com/youtube/vitess/go/vt/vtgate/queryinfo"
+)
+
+// Merge specifies the parameters for a merge primitive
+type Merge struct {
+	Left, Right Primitive `json:",omitempty"`
+}
+
+// Execute performs an execute
+func (merge *Merge) Execute(vcursor VCursor, queryConstruct *queryinfo.QueryConstruct, joinvars map[string]interface{}, wantfields bool) (*sqltypes.Result, error) {
+	var lres, rres *sqltypes.Result
+	var lerr, rerr error
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func(vc *VCursor, qc *queryinfo.QueryConstruct, jv map[string]interface{}, wf bool) {
```

Comment:
I agree with using the wg, was more commenting on the duplication of code for the sake of closure-accessing the return value